src/py42/services/_auth.py

```diff
@@ -41,3 +41,12 @@ def _get_credentials(self):
         headers = {"totp-auth": str(current_token)} if current_token else None
         response = self._auth_connection.get(uri, params=params, headers=headers)
         return u"v3_user_token {}".format(response["v3_user_token"])
+
+
+class JWTAuth(C42RenewableAuth):
+    def __init__(self, jwt_provider):
```

Comment:
How will this handle `totp` ?

Reply:
Looking at implementation of `totp` it seems like it does exactly what is needed in this task and the ticket has very specific requirement description so after a lot of thought I just went ahead and did something so that it can be discussed.

Reply:
if your custom method of getting a JWT requires 2FA, that will need to be handled as part of the custom method you supply.